File: Day3/guess_the_number.rb
def main
    lives_remaining = 3
    number_to_guess = rand(0..10)
  
    puts "Secret number generated. Guess it!"
  
    while lives_remaining > 0
      puts "#{lives_remaining} lives remaining"
      puts "Enter a number: "
      num = gets.chomp.to_i  # Convert input to integer
  
      if num == number_to_guess
        puts "You've won!"
        break
      else
        puts "Incorrect. Try again."
        lives_remaining -= 1
      end
    end
  
    if lives_remaining == 0
      puts "You've lost. The number was #{number_to_guess}."
    end
  end
  
main  # Call the main function to start the game
  

File: 3_guess_the_number/guess_the_number.go
package main

import (
	"fmt"
	"math/rand"
)

func play_game(lives_remaining int)(bool, error)  {
	var number_to_guess int = rand.Intn(10)
	println("You have ", lives_remaining, " lives. Good luck!")
	for lives_remaining > 0{
		println("Enter the number: ")
		var num int
		_, err := fmt.Scanln(&num)
		if err != nil {
			println("That's not a valid number! Please try again.")
			continue // Skip the rest of the loop and start from the beginning
		}

		if num == number_to_guess {
			println("You won!")
			
			return true, nil
		}
		lives_remaining -= 1
		if lives_remaining == 0 {
			print("Sorry. You lose!")
			break
		}
        print("Try again! You have ", lives_remaining, " lives remaining.")
	}
	return false, nil

}

func main(){
	var lives_remaining int = 3
    play_game(lives_remaining)
}